client/components/Planning/PlanningEditor/index.jsx

```diff
@@ -20,20 +20,87 @@ import {ToggleBox} from '../../UI';
 import {PlanningEditorHeader} from './PlanningEditorHeader';
 import {CoverageArrayInput} from '../../Coverages';
 import {EventMetadata} from '../../Events';
+import {ITEM_TYPE} from '../../../constants';
+import {stripHtmlRaw} from 'superdesk-core/scripts/apps/authoring/authoring/helpers';
 
 export class PlanningEditorComponent extends React.Component {
     constructor(props) {
         super(props);
 
         this.dom = {slugline: null};
         this.onChange = this.onChange.bind(this);
+        this.createNewPlanningFromNewsItem = this.createNewPlanningFromNewsItem.bind(this);
+    }
+
+    componentWillMount() {
+        // If we are creating a new planning item for 'add-to-planning'
+        if (this.props.addNewsItemToPlanning) {
```

Comment:
Can you simplify the `if` condition over here i.e. remove the nesting.

Reply:
I think each nest of 'if' statement is required here because it meets those conditions. This will however change when I try to resolve merge conflicts during coverage item actions in another PR.

Reply:
```
if (!this.props.addNewsItemToPlanning) {
    return
}
if (!get(this.props, 'item._id')) {
    const newPlanning = this.createNewPlanningFromNewsItem();
    this.props.onChangeHandler(null, newPlanning);
} else if (get(this.props, 'item._addNewCoverage')) {
    // As per current logic
}
```

Reply:
Per my comment, I meant to update it in another PR when I resolve the mrege conflicts as this file will change again during rebase - not that it is not doable or should not be done. Anyways, I will update it here. Cheers!